src/tools/ScenarioMeasurement/Startup/TimeToMain2Parser.cs

```diff
@@ -0,0 +1,108 @@
+﻿using Microsoft.Diagnostics.Tracing;
+using Microsoft.Diagnostics.Tracing.Parsers;
+using Reporting;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace ScenarioMeasurement;
+
+public class TimeToMain2Parser : IParser
+{
+    private const string EventSourceName = PerfLabValues.EventSourceName;
+    private const string OnMainEventName = PerfLabValues.OnMainEventName;
+    private const int OnMainEventId = PerfLabValues.OnMainEventId;
+
+    public void EnableKernelProvider(ITraceSession kernel)
+    {
+        kernel.EnableKernelProvider(TraceSessionManager.KernelKeyword.Process, TraceSessionManager.KernelKeyword.Thread, TraceSessionManager.KernelKeyword.ContextSwitch);
+    }
+
+    public void EnableUserProviders(ITraceSession user)
+    {
+        user.EnableUserProvider(EventSourceName, TraceEventLevel.Verbose);
+    }
+
+    public IEnumerable<Counter> Parse(string mergeTraceFile, string processName, IList<int> pids, string commandLine)
+    {
+        var results = new List<double>();
+        var threadTimes = new List<double>();
+        double threadTime = 0;
+        var ins = new Dictionary<int, double>();
+        double start = -1;
+        int? pid = null;
+        using (var source = new TraceSourceManager(mergeTraceFile))
+        {
+            source.Kernel.ProcessStart += evt =>
+            {
+                if (!pid.HasValue && ParserUtility.MatchProcessStart(evt, source, processName, pids, commandLine))
+                {
+                    pid = evt.ProcessID;
+                    start = evt.TimeStampRelativeMSec;
+                }
+            };
+
+            source.Source.Kernel.ThreadCSwitch += evt =>
```

Comment:
Do we know get CSwitch events on Linux? Otherewise, shouldn't this go inside the `IsWindows` check?

Reply:
We don't. But it's also doing nothing on Linux, so I left it without check, just in case it starts to work in the future.